spacegrep/src/bin/Spacegrep_main.ml

```diff
@@ -23,6 +25,7 @@ type config = {
   pattern: string option;
   pattern_files: string list;
   doc_files: string list;
+  timeout: int option;
```

Comment:
Note that the semgrep-core accepts actually a float, so you can timeout at less than 1s, but then it's more complicated, you can't use alarm, you need to use a more specialized API (see semgrep-core/cli/Main.ml timeout function).
Note also that in semgrep-core the timeout is per processed file, not the full lenght of the program run.

Maybe you can add a comment that the int unit is in seconds.